src/redux/NewPostSlice.tsx

```diff
@@ -0,0 +1,73 @@
+import { collection, addDoc } from "firebase/firestore";
+import { db } from "../firebase";
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import { ref, getDownloadURL, getStorage, uploadBytes } from "firebase/storage";
+import slugify from "@sindresorhus/slugify";
+
+interface AddNewPostType {
+  title: string;
+  photo: File;
+  description: string;
+  firstName: string;
+  lastName: string;
+  updatedBy: string;
+  profilePhotoPath: string;
+}
+interface NewPostState {
+  status: string;
+}
+const initialState: NewPostState = {
+  status: "idle",
+};
+export const AddNewPost = createAsyncThunk(
+  "AddNewPost",
+  async (data: AddNewPostType) => {
+    const slug = slugify(data.title, {
+      lowercase: false,
+    });
+    const timestamp = new Date().toISOString();
+    const storage = getStorage();
+    const storageRef = ref(
+      storage,
+      `Posts/${data.updatedBy}/postphotos${timestamp}`
+    );
+    await uploadBytes(storageRef, data.photo);
+    const downloadURL = await getDownloadURL(storageRef);
+    const postData = {
+      title: data.title,
+      description: data.description,
+      slug: slug,
+      createdAt: timestamp,
+      updatedAt: timestamp,
+      updatedBy: data.updatedBy,
+      photo: downloadURL,
+      firstName: data.firstName,
+      lastName: data.lastName,
+      profilePhotoPath: data.profilePhotoPath,
```

Comment:
can't we directly ass a filed user and in user: {fname, lname,profilephoto}? @Rishab-innovative